tests/test_lab_init.py

```diff
@@ -22,26 +18,54 @@ class TestLabInit:
     @patch("instructlab.config.init.Repo.clone_from")
     def test_init_noninteractive(self, mock_clone_from):
         runner = CliRunner()
-        with runner.isolated_filesystem():
+        with runner.isolated_filesystem() as td:
+            os.environ["HOME"] = td
+            # lazily import this module as a result of the following issue:
+            # https://github.com/instructlab/instructlab/issues/1582
+            # First Party
+            from instructlab import lab
+            from instructlab.configuration import DEFAULTS
+
+            DEFAULTS._reset()
```

Comment:
I suggest to move `DEFAULTS._reset()` to the `cli_runner` fixture.

Reply:
Is this a blocking requested change or can we move it to an issue? There are other PRs that are blocked waiting on this one.

Reply:
This comment is merely a suggestion.

The removal of `cli_runner` fixture and assignment to `os.environ` are blockers. I created https://github.com/instructlab/instructlab/pull/1617 specifically to solve the problem with leaking env vars.

Reply:
Fixed this in commit https://github.com/instructlab/instructlab/pull/1471/commits/dca8b1bba3fd3c044cc9f62ef1827a8df4d75517